src/toil/server/wes/tasks.py

```diff
@@ -0,0 +1,279 @@
+    def __init__(self, work_dir: str, request: Dict[str, Any], engine_options: List[str]):
+        self.work_dir = work_dir
+        self.request = request
+        self.engine_options = engine_options
+
+        self.wf_type: str = request["workflow_type"].lower().strip()
+        self.version: str = request["workflow_type_version"]
+
+        self.exec_dir = os.path.join(self.work_dir, "execution")
+        self.out_dir = os.path.join(self.work_dir, "outputs")
+
+        self.default_job_store = "file:" + os.path.join(self.work_dir, "toil_job_store")
+        self.job_store = self.default_job_store
+
+    def write(self, filename: str, contents: str) -> None:
+        with open(os.path.join(self.work_dir, filename), "w") as f:
+            f.write(contents)
+
+    def get_state(self) -> str:
+        with open(os.path.join(self.work_dir, "state"), "r") as f:
+            return f.read()
+
+    def set_state(self, state: str) -> None:
+        self.write("state", state)
+
+    def write_workflow(self, src_url: str) -> str:
+        """
+        Fetch the workflow file from its source and write it to a destination
+        file. The destination is returned.
+        """
+        logger.info(f"Processing workflow_url: '{src_url}'...")
+        dest = src_url
+
+        if src_url.startswith("file://"):
+            logger.info(f"Linking workflow from filesystem.")
+            dest = os.path.join(self.exec_dir, os.path.basename(src_url))
+            link_file(src=src_url[7:], dest=dest)
+        elif src_url.startswith(("http://", "https://")):
+            logger.info(f"Downloading workflow_url from the Internet.")
+            # TODO: Download workflow files from the Internet
+            raise NotImplementedError
+        else:
+            logger.info(f"Using workflow from relative URL.")
+            dest = os.path.join(self.exec_dir, src_url)
+
+        return dest
+
+    def sort_options(self) -> List[str]:
+        """
+        Sort the command line arguments in the order that can be recognized by
+        the workflow execution engine.
+        """
+        options = []
+
+        # First, we pass the default engine parameters
+        options.extend(self.engine_options)
+
+        # Then, we pass the user options specific for this workflow run. This should override the default
+        for key, value in self.request.get("workflow_engine_parameters", {}).items():
+            if value is None:  # flags
+                options.append(key)
+            else:
+                options.append(f"{key}={value}")
+
+        # determine job store and set a new default if the user did not set one
+        cloud = False
+        for option in options:
+            if option.startswith("--jobStore="):
+                self.job_store = option[11:]
+                options.remove(option)
+            if option.startswith(("--outdir=", "-o=")):
+                options.remove(option)
+        if self.job_store.startswith(("aws", "google", "azure")):
```

Comment:
You might need a trailing `:` here. Or better yet, use the same `parseLocator` function as is called here: https://github.com/DataBiosphere/toil/blob/aa50bbfdef66bd9a861fb889325e476405fe25b6/src/toil/common.py#L903-L914

Reply:
Good catch!